lib/ex_doc/formatter/html/autolink.ex

```diff
@@ -562,7 +565,7 @@ defmodule ExDoc.Formatter.HTML.Autolink do
         nil
 
       path ->
-        path = List.to_string(path)
```

Comment:
this was the culprit, when inspecting variables we had:

```
lib_dirs: [
  {"/Users/wojtek/src/elixir-lang/elixir/lib/elixir/ebin",
   "https://hexdocs.pm/elixir/"},
  {"/Users/wojtek/src/elixir-lang/elixir/lib/eex/ebin",
   "https://hexdocs.pm/eex/"},
  {"/Users/wojtek/src/elixir-lang/elixir/lib/iex/ebin",
   "https://hexdocs.pm/iex/"},
  {"/Users/wojtek/src/elixir-lang/elixir/lib/logger/ebin",
   "https://hexdocs.pm/logger/"},
  {"/Users/wojtek/src/elixir-lang/elixir/lib/mix/ebin",
   "https://hexdocs.pm/mix/"},
  {"/Users/wojtek/src/elixir-lang/elixir/lib/ex_unit/ebin",
   "https://hexdocs.pm/ex_unit/"},
  {"/usr/local/opt/asdf/installs/erlang/21.1/lib",
   "http://www.erlang.org/doc/man/"}
]
path: "/Users/wojtek/src/elixir-lang/elixir/bin/../lib/mix/ebin/Elixir.Mix.Tasks.Compile.App.beam"
```

which explains the discrepancy.

I couldn't easily write a test for it, verified by regenerating Elixir docs.